build.js

```diff
@@ -1,88 +1,269 @@
 #!/usr/bin/env node
 
 import child_process from 'child_process';
+import fs from 'fs';
+import os from 'os';
 import path from 'path';
 import process from 'process';
 
+import { getFiles, getTestFiles, all_subdirs } from './files.js';
+
+const production = process.env.NODE_ENV === 'production';
+const useWasm = os.arch() != 'x64';
+
 // ensure node_modules is present and up to date
 child_process.spawnSync('tools/node-modules', ['make_package_lock_json'], { stdio: 'inherit' });
 
+// List of directories to use when resolving import statements
+const nodePaths = ['pkg/lib'];
+
+// context options for distributed pages in dist/
+const pkgOptions = {
+    ...!production ? { sourcemap: "external" } : {},
+    bundle: true,
+    external: ['*.woff', '*.woff2', '*.jpg', '*.svg', '../../assets*'], // Allow external font files which live in ../../static/fonts
+    legalComments: 'external', // Move all legal comments to a .LEGAL.txt file
+    loader: {
+        ".js": "jsx",
+        ".py": "text",
+        ".sh": "text",
+    },
+    minify: production,
+    nodePaths,
+    outbase: './pkg',
+    outdir: "./dist",
+    target: ['es2020'],
+};
+
+// context options for qunit tests in qunit/
+const qunitOptions = {
+    bundle: true,
+    minify: false,
+    nodePaths,
+    outbase: './pkg',
+    outdir: "./qunit",
+    loader: {
+        ".sh": "text",
+    },
+};
+
 const parser = (await import('argparse')).default.ArgumentParser();
 parser.add_argument('-c', '--config', { help: "Path to webpack.config.js", default: "webpack.config.js" });
```

Comment:
Let's wait for @mvollmer 's review, as this is otherwise green, and rather "expensive" (pixel tests, node_modules rebuild). I want to do a follow-up anyway to simplify `tests` in files.js, happy to clean it up then. Of course, when Marius finds other issues, we can fix this right away.